genesis_devtools/libvirt.py

```diff
@@ -227,12 +236,28 @@ def has_net(name: str) -> bool:
 
 def destroy_domain(name: str) -> None:
     """Delete domain."""
-    subprocess.run(
-        f"sudo virsh destroy {name} 1>/dev/null && "
-        f"sudo virsh undefine {name} 1>/dev/null",
-        shell=True,
-        check=True,
-    )
+    disk_path = get_domain_disk(name)
+
+    try:
+        subprocess.run(
+            f"sudo virsh destroy {name} 1>/dev/null && "
+            f"sudo virsh undefine {name} 1>/dev/null",
+            shell=True,
+            check=True,
+        )
+    except subprocess.CalledProcessError:
```

Comment:
Minor: we may catch except subprocess.CalledProcessError and just log it
, but try to undefine in `finally`, so it'll be less code (we `undefine` VM in both cases)